### PR TITLE
Remove array in GamePadDPad constructor

### PR DESCRIPTION
**Pull request by 06needhamt**
This PR fixes issue #4910

### CODE REVIEW

File: MonoGame.Framework/Input/GamePadDPad.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public GamePadDPad(ButtonState upValue, ButtonState downValue, ButtonState leftV
             Right = rightValue;
         }
 
-        internal GamePadDPad(params Buttons[] buttons)
+        internal GamePadDPad(Buttons[] buttons)
```
Comment (tomspilman) [inline MonoGame.Framework/Input/GamePadDPad.cs]: @06needhamt - I think I misled you a bit in https://github.com/mono/MonoGame/issues/4910.  We need to remove the array completely...  there is no need for any array here.  The function should be `internal GamePadDPad(Buttons buttons)`.

This of course means the `foreach` loop is removed with it just testing the single `buttons` type.

**Comment by 06needhamt** [comment]
Ah i see will change it now

**Comment by 06needhamt** [comment]
does the game pad state constructor need to be changed too or should it take buttons[0]?

**Comment by tomspilman** [comment]
> does the GamePad State constructor need to be changed too or should it take buttons[0]?

Yes...  it should be changed too.  There is no need for any array passed or used in the `GamePadDPad` constructor.

**Comment by 06needhamt** [comment]
Everything is fixed now i will create a new PR so there is only one commit

**Comment by 06needhamt** [comment]
Im not sure what happened here i rebased the PR but it is still showing the individual commits

**Comment by KonajuGames** [comment]
I've always found rebasing to be a black magic thing that I can rarely get right.  We can squash the commits when we merge the PR.

**Comment by 06needhamt** [comment]
ok thanks a lot this is my first contribution so i was unfamiliar witth the codebase hence so many commits. also i have errors on certain platforms when trying to build on my machine so i am unable to test all platforms

**Comment by tomspilman** [comment]
This looks to be working...  merging!

Thanks @06needhamt !

**Comment by nkast** [comment]
This one broke XNA compatibility.
FarseerPhysics samples was using it for Virtual Gamepad.

**Comment by tomspilman** [comment]
Must be an undocumented ctor.  I'm fine with adding it back as long as the internal code doesn't use it.

**Comment by Halofreak1990** [comment]
> Must be an undocumented ctor. I'm fine with adding it back as long as the internal code doesn't use it.
Just checked the XNA documentation for the [GamePadDPad](https://msdn.microsoft.com/en-us/library/microsoft.xna.framework.input.gamepaddpad.gamepaddpad(v=xnagamestudio.40).aspx) structure, and it doesn't have a constructor taking an array. Never had (a public) one, either.

**Comment by nkast** [comment]
@Halofreak1990 

The PR changed the ctor of  `GamePadState` as well.
https://msdn.microsoft.com/en-us/library/bb975283.aspx

It changed the Test for GamePadState as well, which probably broke the Tests for XNA (or should have).

**Pull request merged**